Lib/test/test_bytes.py

```diff
@@ -2519,6 +2516,34 @@ def check(funcs, a=None, *args):
         check([clear] + [upper] * 10, bytearray(b'a' * 0x400000))
         check([clear] + [zfill] * 10, bytearray(b'1' * 0x200000))
 
+    @unittest.skipUnless(support.Py_GIL_DISABLED, 'this test can only possibly fail with GIL disabled')
+    @threading_helper.reap_threads
+    @threading_helper.requires_working_threading()
+    def test_free_threading_bytearrayiter(self):
+        # Non-deterministic but good chance to fail if bytearrayiter is not free-threading safe.
+        # We are fishing for a "Assertion failed: object has negative ref count".
+
+        def iter_next(b, it):
+            b.wait()
+            list(it)
+
```

Comment:
let's add one test for reduce and set state as well?

Reply:
They can't really fail.

Reply:
They can't fail but if there would be a race then tsan would catch it.